tests/unit/test_masks.py

```diff
@@ -153,6 +154,17 @@ def it_knows_its_mask_tile(self, request):
 
         assert binary_mask == expected_mask
 
+    def it_knows_custom_image_filters(self):
+        tile = Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, None, None)
+
+        # Create a tissue mask which should not cover anything
+        tissue_mask = TissueMask(
+            image_filters.RgbToGrayscale(), image_filters.OtsuThreshold()
+        )
+
+        binary_mask = tissue_mask(tile)
+        assert not np.any(binary_mask)
```

Comment:
This test smells to me. Here you are not testing the functionality. A unit test need to mock the objects involved in the actual implementation, here you are asserting something that does not imply nothing.

Reply:
Thank you for the review! To be honest, I could not completely get your point... With the unit test above, I just check that the result after applying the custom pipeline differs from the default (I might do that more explicit, as I rely on the current documentation there...) Testing the filters themselves is already covered in the other tests, is it not?

Reply:
My point is that a unit tests that has to proof that your changes are working. So maybe I should've been more clear, sorry. Given that I'd like to see a unit tests that mocks your objects you introduced. For example how do we know that `_CustomFiltersComposition` has been called properly? Ho can we know that the `setter` within that is doing the right thing? Is it working only on a `Slide` or on a `Tile` too?

My hunch is that the changes of the custom filters works only if applied on a `Slide` and not on a `Tile` I'd like to see a tests with this kind of proof.

IMHO i do not think that the test `it_knows_custom_image_filters` is not applying the filters you believe. So to disprove my hypothesis I'd like to see a tests with this kind of assertions.

Hope I've been clear 🤷🏽‍♂️ . Let me know 😄

Reply:
You are of cause entirely right! I will go through the code again. The missing implementation for Tile is almost embarrassing ... 

Is there any documentation regarding your preferred unit tests and mock functions? They are amazingly sophisticated

Reply:
👍🏽 thanks a lot. Anyway there's no official preference doc for unit testing. I'd suggest a quick reading of https://realpython.com/python-mock-library/. Furthermore if you want you can join the slack channel (link in the README) and we can discuss there... or even we can have a brief call and we can do the implementation together. Let me know what do you think

Reply:
Thank you very much for the additional information! I will join the Slack channel which seems like a valuable resource. 

I used my spare time on the train and added both the missing implementation and (hopefully) proper unit tests. 

What do you think?